File: post-service/src/main/java/com/dislinkt/postservice/service/PostService.java
package com.dislinkt.postservice.service;

import com.dislinkt.postservice.model.Comment;
import com.dislinkt.postservice.model.Post;

import java.util.List;

public interface PostService {
    void save(Post post);
    List<Post> findUserPosts(String userId);

    List<Post> getFeed(String id) throws InterruptedException;

    int likePost(String postID, String userId);
    int dislikePost(String postID, String userId);

    void comment(String postID, Comment comment);

    Post getPost(String postID);
    List<Post> getPublicFeed() throws InterruptedException;

}

File: frontend/src/app/modules/post/services/post.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { AuthService } from 'src/app/core/auth.service';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'any'
})
export class PostService {

  baseUrl: string = environment.baseUrl + 'post-service/posts/';

  constructor(private http: HttpClient, private authService: AuthService) { }

  likePost(postId: string): Observable<any> {
    return this.http.post<any>(this.baseUrl + postId + '/like', { headers: this.authService.getHeader() })
  }

  dislikePost(postId: string): Observable<any> {
    return this.http.post<any>(this.baseUrl + postId + '/dislike', { headers: this.authService.getHeader() })
  }

  createPost(statusText: string, image: any): Observable<any> {
    let post = { content: statusText, base64Image: image };
    return this.http.post(this.baseUrl, post, { headers: this.authService.getHeader() })
  }

  postComment(postId: string, commentText: string): Observable<any> {
    let comment = { userId: this.authService.getUserId(), comment: commentText }
    return this.http.post(this.baseUrl + postId + '/comment', comment, { headers: this.authService.getHeader() })
  }
}
